qa/L0_backend_python/lifecycle/lifecycle_test.py

```diff
@@ -207,6 +233,13 @@ def test_infer_pymodel_error(self):
                     self.assertTrue(
                         False, "Wrong exception raised or did not raise an exception"
                     )
+        expected_count_increase = 1
+        self._assert_metrics(
+            model_name,
+            "BACKEND",
```

Comment:
Please (a) update the Failure Count row in https://github.com/triton-inference-server/server/blob/main/docs/user_guide/metrics.md to mention the current set of supported `reasons` and (b) add a small section describing each of the reasons in more detail and when they would be expected to be reflect. For example, "REJECTED" and "BACKEND" may not be immediately clear without explanation.

Reply:
(c) please also mention the caveat about ensemble metrics all being `OTHER` for now in that docs section, similar to this comment: https://github.com/triton-inference-server/server/pull/7423/files#r1673381047